tests/string_test.py

```diff
@@ -644,11 +644,17 @@ def test_encoding(self):
         result = s1.idna_encode()
         self.assertListEqual([i.encode("idna").decode("ascii") for i in a1], result.to_list())
 
-        a2 = ['xn--mnchen-3ya', 'xn--zrich-kva', ' xn--zrich-boguscode', 'xn--!!']
+        # validate encoding with empty string
+        e = ["", "abc", "", "123"]
+        ak_e = ak.array(e)
+        result = ak_e.idna_encode()
+        self.assertListEqual([i.encode("idna").decode("ascii") for i in e], result.to_list())
+
+        a2 = ['xn--mnchen-3ya', 'xn--zrich-kva', ' xn--zrich-boguscode', 'xn--!!', 'example.com']
         s2 = ak.array(a2)
         result = s2.idna_decode()
         # using the below assertion due to a bug in `Strings.to_ndarray`. See issue #1828
-        self.assertListEqual(["münchen", "zürich", "", ""], result.to_list())
+        self.assertListEqual(["münchen", "zürich", "", "", "example.com"], result.to_list())
```

Comment:
I don't know much about encoding, but should we be thinking about the situation where we get an ascii string that happens to start with `xn--`? I think either way this covers everything our users have asked for so i think it's good